src/FormCollection/Resources/assets/src/controller.ts

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    static targets = ['entry'];
+
+    static values = {
+        allowAdd: Boolean,
+        allowDelete: Boolean,
+        buttonAdd: String,
+        buttonDelete: String,
+        prototypeName: String,
+        prototype: String,
+        startIndex: Number,
+    };
+
+    allowAddValue: boolean;
+    allowDeleteValue: boolean;
+    buttonAddValue: string;
+    buttonDeleteValue: string;
+    prototypeNameValue: string;
+    prototypeValue: string;
+    startIndexValue: number;
+
+    /**
+     * Number of elements for the index of the collection
+     */
+    index = 0;
+
+    controllerName: string;
+
+    entryTargets: Array<any> = [];
+
+    connect() {
+        this.controllerName = this.context.scope.identifier;
+        this.index = this.startIndexValue ? this.startIndexValue : this.entryTargets.length - 1;
+
+        if (!this.prototypeNameValue) {
+            this.prototypeNameValue = '__name__';
+        }
+
+        this._dispatchEvent('form-collection:pre-connect', {
+            allowAdd: this.allowAddValue,
+            allowDelete: this.allowDeleteValue,
+        });
+
+        if (this.allowAddValue) {
+            // Add button Add
+            const buttonAdd = this._textToNode(this.buttonAddValue);
+            this.element.prepend(buttonAdd);
```

Comment:
I think this should actually be added to the *bottom*. In the future, we may make it configurable, bottom seems like a better default to me.